tests/unit/test_configs.py

```diff
@@ -156,3 +159,27 @@ def test_load_dash_env_vars_refects_to_os_environ(empty_environ):
 def test_app_name_server(empty_environ, name, server, expected):
     app = Dash(name=name, server=server)
     assert app.config.name == expected
+
+
+@pytest.mark.parametrize(
+    "prefix, partial_path, expected",
+    [
+        ("/", "/", "/"),
+        ("/my-dash-app", "/", "/my-dash-app/"),
+
+        ("/", "/page-1", "/page-1"),
+        ("/my-dash-app", "/page-1", "/my-dash-app/page-1"),
+
+        ("/", "/page-1/", "/page-1/"),
+        ("/my-dash-app", "/page-1/", "/my-dash-app/page-1/"),
+
+        ("/", "/page-1/sub-page-1", "/page-1/sub-page-1"),
+        ("/my-dash-app", "/page-1/sub-page-1", "/my-dash-app/page-1/sub-page-1"),
+
+        ("/", "relative-page-1", "relative-page-1"),
+        ("/my-dash-app", "/my-dash-apprelative-page-1", "/my-dash-apprelative-page-1"),
```

Comment:
These relative URLs are ambiguous in this scenario as we don't know what page we're on. 

For example, if we were on page `/my-dash-app/this-page` and then provided a `relative-page-1`, the resolved URL should be `/my-dash-app/this-page/relative-page-1` but we can't provide that to the front end because we are unaware of `this-page`.

So, I thought we'd just keep it a simple prefix without any additional logic.

Relative URLs aren't documented in Dash nor used very much anyway.